tracer/src/Datadog.Trace/Sampling/SamplingMechanism.cs

```diff
@@ -72,4 +74,27 @@ internal static class SamplingMechanism
     /// The only available sampling priority is <see cref="SamplingPriorityValues.UserKeep"/> (2).
     /// </summary>
     public const int SpanSamplingRule = 8;
+
+    public static string GetTraceContextDecisionMakerTagValue(int mechanism)
+    {
+        // set the sampling mechanism trace tag
+        // * only set tag if priority is AUTO_KEEP (1) or USER_KEEP (2)
+        // * do not overwrite an existing value
+        // * don't set tag if sampling mechanism is unknown (null)
+        // * the "-" prefix is a left-over separator from a previous iteration of this feature (not a typo or a negative sign)
+
+        return mechanism switch
```

Comment:
Not a `string[]` with `mechanism` as the index into the array? 😝

Reply:
and have an allocation for that!!??? hahahaha, I mean is still a small switch so we can avoid the array for now.

Reply:
I was half-kidding, but I wouldn't assume anything without benchmarking. It's a one-time array allocation and then single lookups into an array vs possibly multiple compares in the switch. Also smaller IL?

Would it help to at least the most common ones to the top? (just curious, no need to change anything in this PR)